components/formats-gpl/src/loci/formats/in/NativeND2Reader.java

```diff
@@ -2412,9 +2412,10 @@ private void parseText(String textString, int offsetCount, boolean useDimensions
 
       lines = textString.split(" ");
       for (int i=0; i<lines.length; i++) {
-        String key = lines[i++];
-        while (!key.endsWith(":") && key.indexOf('_') < 0 && i < lines.length) {
-          key += " " + lines[i++];
+        StringBuffer sb = new StringBuffer(lines[i++]);
+        while (sb.charAt(sb.length() - 1) != ':' && sb.indexOf("_") < 0 && i < lines.length) {
```

Comment:
This seems to have broken the `setId` call on some nd2 datasets - see https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-DEV-merge-full-repository/703/console